runtime/compiler/x/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -123,34 +123,30 @@ uint8_t *J9::X86::AheadOfTimeCompile::initializeAOTRelocationHeader(TR::Iterated
    TR_J9VMBase *fej9 = (TR_J9VMBase *)(_cg->fe());
    TR_SharedCache *sharedCache = fej9->sharedCache();
    TR::Compilation *comp = _cg->comp();
-   TR::SymbolValidationManager *symValManager = comp->getSymbolValidationManager();
-   uintptr_t numTrampolines;
-   TR::LabelSymbol *label;
-   TR_ResolvedMethod *resolvedMethod;
-   TR_VirtualGuard *guard;
-   uint8_t flags = 0;
-   uint8_t *cursor = relocation->getRelocationData();
-
    TR_RelocationRuntime *reloRuntime = comp->reloRuntime();
    TR_RelocationTarget *reloTarget = reloRuntime->reloTarget();
+   uintptr_t numTrampolines;
 
-   // size of relocation goes first in all types
-   *(uint16_t *)cursor = relocation->getSizeOfRelocationData();
-   cursor += 2;
-   uint8_t  wideOffsets = relocation->needsWideOffsets() ? RELOCATION_TYPE_WIDE_OFFSET : 0;
-   *cursor++ = (uint8_t)relocation->getTargetKind();
-   uint8_t *flagsCursor = cursor++;
-   *flagsCursor = wideOffsets;
-   uint32_t *wordAfterHeader = (uint32_t*)cursor;
-#if defined(TR_HOST_64BIT)
-   cursor += 4; // padding
-#endif
+   uint8_t *cursor         = relocation->getRelocationData();
+   uint8_t targetKind      = relocation->getTargetKind();
+   uint16_t sizeOfReloData = relocation->getSizeOfRelocationData();
+
+   // Zero-initialize header
+   memset(cursor, 0, sizeOfReloData);
 
-   // This has to be created after the kind has been written into the header
    TR_RelocationRecord storage;
-   TR_RelocationRecord *reloRecord = TR_RelocationRecord::create(&storage, reloRuntime, reloTarget, reinterpret_cast<TR_RelocationRecordBinaryTemplate *>(relocation->getRelocationData()));
+   TR_RelocationRecord *reloRecord = TR_RelocationRecord::create(&storage, reloRuntime, targetKind, reinterpret_cast<TR_RelocationRecordBinaryTemplate *>(cursor));
+
+   uint8_t wideOffsets = relocation->needsWideOffsets() ? RELOCATION_TYPE_WIDE_OFFSET : 0;
+   uint32_t *wordAfterHeader = &reinterpret_cast<TR_RelocationRecordPicTrampolineBinaryTemplate *>(cursor)->_numTrampolines;
```

Comment:
Temporary, this is removed by the end of the PR. Consequence of the hacky
```
#if defined(TR_HOST_64BIT)
   uint32_t _extra;
#endif
```
field in `TR_RelocationRecordBinaryTemplate`